Terminal.Gui/Views/Scroll/Scroll.cs

```diff
@@ -176,7 +176,9 @@ protected internal override bool OnMouseEvent (MouseEvent mouseEvent)
 
         if (mouseEvent.Flags.HasFlag (MouseFlags.Button1Pressed) && location < sliderPos.start)
         {
-            Position = Math.Max (Position - barSize, 0);
+            int distance = sliderPos.start - location;
+            int scrollAmount = (int)((double)distance / barSize * (Size - barSize));
+            Position = Math.Max (Position - scrollAmount, 0);
```

Comment:
As I said, in the `Scroll` must only navigate one page at a time. The problem was on the `ScrollSlider` class and calculations fixes for the `KeepContentInAllViewport` property. Now the synchronization unit test are failing but ran the scenario better.